Building Forms with Ruby on Rails

emasdev opened the pull request:
In this project, we built a form the old fashioned way and then the Rails way

- Generated the model/controller/migrations for Users.
- Validate input data from form new/edit User form.

emasdev committed:
Rails start project with Users functionality

- Add User (Model/Controller/Routes)
- Add user/new view

MarilenaRoque committed:
Create form and set the controllers

Created an html form to create new users
Set up params method to receive the user params
Set a hash to retrieve all the user params

emasdev committed:
Fix password input name in user/new view

emasdev committed:
Update user/new form

- change user/new form using form_tag helper
- change user/new form using form_for helper

MarilenaRoque committed:
Set up edit feature

Build methods and controllers for #eit and #update
Build view for edit
Set display error messages for invalid inputs

MarilenaRoque committed:
Set errors message for new view

emasdev committed:
Add documentation and final steps

- Build readme
- Add stickler
- Fix rubocop offenses

mikenath223 reviewed (changes requested):
### Hi Team 👋

You've both done well on this.

However, kindly check the commented code below for changes required. 👇

In addition, kindly fix these issues and request a new code review.

**HAPPY CODING ✨**

mikenath223 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,36 @@
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
- [x] Please remove the commented code.

mikenath223 commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,52 @@
+<% end %>
+
+<!--
+<%= form_tag("/users", method: "post") do %>
```
- [x] Kindly use the users_path helper here instead.

mikenath223 commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,52 @@
+-->
+
+<!--
+<form action="/users" method="post" accept-charset="UTF-8">
```
- [x] Use the users_path helper here too.

mikenath223 commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,52 @@
+
+<!--
+<%= form_tag("/users", method: "post") do %>
+  <%= label_tag(:username, "Name:") %>
+  <%= text_field_tag(:username) %> <br><br>
+    <%= label_tag(:email, "Email:") %>
+  <%= email_field_tag(:email) %> <br><br>
+    <%= label_tag(:password, "Password:") %>
+  <%= password_field_tag(:password) %> <br><br>
```
- [x] For each of the label tag here please change it to this format:  `<%= email_field_tag 'email' %>`

mikenath223 commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,52 @@
+<h2>Create User</h2>
+
+<%= form_for @user, url: {action: "create"} do |f| %>
```
- [x]  No need to add a url here the @user variable will be enough.

MarilenaRoque committed:
Fixing TSE Requirements

- Used heper _path on new.html.erb
- Removed commented line
- Aply best practive to form fields

MarilenaRoque commented:
## Hello @mikenath223. Thanks for the review! 

We're happy to say that all the requirements were applied:

- [x] Used _path helper 

 - [x] Removed the commented code.

- [x] For each of the label tag we changed it to this format: <%= email_field_tag 'email' %>

- [x] Removed url from form_for tag.

Dandush03 commented:
# Status: :rocket: Project Approved :rocket: 
## Great job applying does changes
# Happy Coding! :computer: :nerd_face:
